app/controllers/wishlists_controller.rb

```diff
@@ -21,6 +21,11 @@ def create
     end
   end
 
+  def show
+    @wishlist = Wishlist.created_by(current_user).find_by_id(params[:id])
```

Comment:
We'll get redirected to root as a rescue from `CanCan::AccessDenied` (as set in application_controller.rb).